susan/apps/dhcp/dhcp.py

```diff
@@ -94,13 +107,14 @@ def identify_packet(pkt):
         dhcp_pkt = pkt.get_protocol(dhcp.dhcp)
         request_type = ''
         opts = dhcp_pkt.options.option_list
+        msg_type = dhcp_const.OPTIONS.MESSAGE_TYPE
+        discover = struct.pack('!B', dhcp_const.REQUEST.DISCOVER)
+        offer = struct.pack('!B', dhcp_const.REQUEST.REQUEST)
         for option in opts:
-            if (option.tag == dhcp_const.OPTIONS.MESSAGE_TYPE and
-                    option.value == struct.pack('B', dhcp_const.REQUEST.DISCOVER)):
+            if option.tag == msg_type and option.value == discover:
                 request_type = REQUEST_MAPPER.DHCPDISCOVER
                 break
-            elif (option.tag == dhcp_const.OPTIONS.MESSAGE_TYPE and
-                  option.value == struct.pack('B', dhcp_const.REQUEST.REQUEST)):
+            elif option.tag == msg_type and option.value == offer:
```

Comment:
what does this mean? If the packet option list contain offer , the request type id DHCPREQUEST why?